File: Data Structures & Algorithms/verifying-an-alien-dictionary/submission-2.go
func isAlienSorted(words []string, order string) bool {
    rank := make(map[byte]int)

    for i:=0;i<len(order);i++{
        rank[order[i]] = i
    }

    for i:=0;i<len(words)-1;i++{
        word1 := words[i]
        word2 := words[i+1]

        foundDiff := true

        for j:=0;j<min(len(word1), len(word2));j++ {
            if word1[j] != word2[j]{
                if rank[word1[j]] > rank[word2[j]]{
                    return false
                }
                foundDiff = false
                break
            }
        }

        if foundDiff && len(word1) > len(word2){
            return false 
        }
        
    }
    return true
}

File: Data Structures & Algorithms/merge-strings-alternately/submission-2.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n1 = word1.size(), n2 = word2.size();
        string s = "";
        int j =0;
        for(int i = 0;i<min(n1,n2);i++){
            s += word1[i];
            s+=word2[j];
            j++;
        }

        if(n1 > n2){
            s.append(word1.begin() + j, word1.end());
        }

        if(n1 < n2){
            s.append(word2.begin() + j, word2.end());
        }

        return s;
    }
};